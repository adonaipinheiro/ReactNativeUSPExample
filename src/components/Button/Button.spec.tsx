import React from 'react';

import { fireEvent, render } from '@testing-library/react-native';

import { Button } from './';

describe('Button', () => {
    it('renders correctly with text', () => {
        const { getByText, toJSON } = render(<Button text="Click me" />);
        const button = getByText('Click me');

        expect(button).toBeTruthy();
        expect(toJSON()).toMatchSnapshot();
    });

    it('calls onPress when pressed', () => {
        const onPressMock = jest.fn();
        const { getByText } = render(
            <Button text="Press" onPress={onPressMock} />
        );

        const button = getByText('Press');
        fireEvent.press(button);

        expect(onPressMock).toHaveBeenCalledTimes(1);
    });

    it('applies the correct styles', () => {
        const { getByText } = render(<Button text="Styled" />);
        const buttonText = getByText('Styled');

        expect(buttonText.props.style).toBeDefined();
    });
});
