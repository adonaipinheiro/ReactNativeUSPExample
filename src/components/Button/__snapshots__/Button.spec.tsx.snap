// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly with text 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#0d3db2",
      "borderRadius": 4,
      "opacity": 1,
      "padding": 10,
      "paddingHorizontal": 24,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  >
    Click me
  </Text>
</View>
`;
